src/wpas/wpa_controller.c

```diff
@@ -820,13 +820,15 @@ __wpa_controller_send_commandf(struct wpa_controller *ctrl, const char *name, ch
 int
 wpa_controller_qrcode(struct wpa_controller *ctrl, const char *dpp_uri, uint32_t *bootstrap_id)
 {
-    char reply[WPA_MAX_MSG_SIZE];
-    size_t reply_length = sizeof reply;
+    char reply[WPA_MAX_MSG_SIZE + 1];
+    size_t reply_length = WPA_MAX_MSG_SIZE;
 
     int ret = wpa_controller_send_commandf(ctrl, "DPP_QR_CODE", reply, &reply_length, "%s", dpp_uri);
     if (ret < 0)
         return ret;
 
+    reply[reply_length] = '\0';
```

Comment:
Terminating the `reply` buffer should be done in `__wpa_controller_send_commandf` such that it doesn't have to be done for each command that calls it.

Reply:
Ok got it.